fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayBanner.kt

```diff
@@ -0,0 +1,115 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Top-level UI for displaying the banner in [TabsTray].
+ *
+ * @param isInMultiSelectMode Boolean indicating whether the tab list is in multi-select mode.
+ * @param onNormalTabsClicked Invoked when the user clicks on the normal tabs button in the default banner.
+ * @param onPrivateTabsClicked Invoked when the user clicks on the private tabs button in the default banner.
+ * @param onSyncedTabsClicked Invoked when the user clicks on the synced tabs button in the default banner.
+ */
+@Composable
+fun TabsTrayBanner(
+    isInMultiSelectMode: Boolean,
+    onNormalTabsClicked: () -> Unit,
+    onPrivateTabsClicked: () -> Unit,
+    onSyncedTabsClicked: () -> Unit,
+) {
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(80.dp),
+    ) {
+        if (isInMultiSelectMode) {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(color = FirefoxTheme.colors.layerAccent),
+                contentAlignment = Alignment.Center,
+            ) {
+                Text(
+                    text = "Multi selection mode",
+                    color = FirefoxTheme.colors.textOnColorPrimary,
+                    style = FirefoxTheme.typography.body1,
+                )
+            }
+        } else {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(all = 16.dp),
+                horizontalArrangement = Arrangement.spacedBy(16.dp),
+            ) {
+                Box(modifier = Modifier.weight(1.0f)) {
+                    PrimaryButton(
+                        text = "Normal tabs",
+                        onClick = onNormalTabsClicked,
+                    )
+                }
+
+                Box(modifier = Modifier.weight(1.0f)) {
+                    PrimaryButton(
+                        text = "Private tabs",
+                        onClick = onPrivateTabsClicked,
+                    )
+                }
+
+                Box(modifier = Modifier.weight(1.0f)) {
+                    PrimaryButton(
+                        text = "Synced tabs",
+                        onClick = onSyncedTabsClicked,
+                    )
+                }
+            }
+        }
```

Comment:
I was admittedly on the fence on this. On one hand, it simplifies the root `TabsTrayBanner` Composable and keeps the logic visible, but then it's an additional Composable leaf node the lambdas have to be passed to. Separately, the lambdas may require a future discussion as they grow to be 5+ in certain Composables. 

What are your thoughts? I'm cool with going that approach for now and then reverting back if the child Composables don't end up being too complex.

Reply:
@gabrielluong, what's your take on making more child Composables here?

Reply:
I don't really have a take on this since it's still a bit early. It makes sense, but I would like to see these boxes be converted to using TabRow and Tab first.

Reply:
This is the ticket where I designated the work for that, as it also includes adding-in the three dot button. I have not added the details (yet) for using `TabRow`.

https://bugzilla.mozilla.org/show_bug.cgi?id=1814991